apps/service_providers/messaging_service.py

```diff
@@ -60,7 +61,13 @@ def s3_client(self):
     def send_whatsapp_text_message(self, message: str, from_number: str, to_number):
         self.client.messages.create(from_=f"whatsapp:{from_number}", body=message, to=f"whatsapp:{to_number}")
 
-    def send_whatsapp_voice_message(self, voice_audio: BytesIO, duration: int, from_number: str, to_number):
+    def send_whatsapp_voice_message(self, synthetic_voice: SynthesizedAudio, from_number: str, to_number):
+        voice_audio = synthetic_voice.audio
+        if synthetic_voice.format != "mp3":
+            voice_audio = audio.convert_audio(
+                synthetic_voice.audio, target_format="mp3", source_format=synthetic_voice.format
+            )
```

Comment:
what about moving this to the SynthesizedAudio class so that you can call:

```python
required_formats = [...]  # sorted by preference (if file is not already in a format in the list it will be converted to the first format listed)
synthetic_voice.get_audio_bytes(supported_formats=required_formats)
```

The conversion will happen in the `SynthesizedAudio` class if necessary.

(Supported formats appear to be ACC, MP4, AMR, MPEG, OGG codec = OPUS)

Reply:
https://github.com/dimagi/open-chat-studio/pull/308/commits/989dc8475607c28c8b598a537eec9d4e67ab22c1